lib/Dialect/RTG/IR/RTGOps.cpp

```diff
@@ -552,6 +552,45 @@ LogicalResult MemoryBlockDeclareOp::inferReturnTypes(
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ArrayCreateOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult ArrayCreateOp::verify() {
+  if (!getElements().empty() &&
+      getElements()[0].getType() != getType().getElementType())
+    return emitOpError(
+               "operand types must match array eleement type, expected ")
+           << getType().getElementType() << " but got "
+           << getElements()[0].getType();
```

Comment:
I think you need to check that all the inputs are instances of the element type.